sklearn/preprocessing/tests/test_data.py

```diff
@@ -288,12 +288,88 @@ def test_polynomial_features_csr_X_dim_edges(deg, dim, interaction_only):
+
+        # make sure Error is raised the sample weights greater than 1d
+        sample_weight_notOK = rng.randn(n_samples, 1) ** 2
+        with pytest.raises(ValueError):
+            scaler.fit(X, y, sample_weight=sample_weight_notOK)
+
+
+@pytest.mark.parametrize(['Xw', 'X', 'sample_weight'],
+                         [([[1, 2, 3], [4, 5, 6]],
+                           [[1, 2, 3], [1, 2, 3], [4, 5, 6]],
+                           [2., 1.]),
+                          ([[1, 0, 1], [0, 0, 1]],
+                           [[1, 0, 1], [0, 0, 1], [0, 0, 1], [0, 0, 1]],
+                           np.array([1, 3])),
+                          ([[1, np.nan, 1], [np.nan, np.nan, 1]],
+                           [[1, np.nan, 1], [np.nan, np.nan, 1],
+                            [np.nan, np.nan, 1], [np.nan, np.nan, 1]],
+                           np.array([1, 3])),
+                          ])
+def test_standard_scaler_sample_weight(Xw, X, sample_weight):
+    # weighted StandardScaler
+    yw = np.ones(len(Xw))
+    scaler_w = StandardScaler()
+    scaler_w.fit(Xw, yw, sample_weight=sample_weight)
+
+    # unweighted, but with repeated samples
+    y = np.ones(len(X))
+    scaler = StandardScaler()
+    scaler.fit(X, y)
+
+    X_test = [[1.5, 2.5, 3.5], [3.5, 4.5, 5.5]]
+
+    assert_almost_equal(scaler.mean_, scaler_w.mean_)
+    assert_almost_equal(scaler.var_, scaler_w.var_)
+    assert_almost_equal(scaler.transform(X_test), scaler_w.transform(X_test))
+
+
+@pytest.mark.parametrize(['Xw', 'X', 'sample_weight'],
+                         [([[0, 0, 1, np.nan, 2, 0],
+                            [0, 3, np.nan, np.nan, np.nan, 2]],
+                           [[0, 0, 1, np.nan, 2, 0],
+                            [0, 0, 1, np.nan, 2, 0],
+                            [0, 3, np.nan, np.nan, np.nan, 2]],
+                           [1., 2.]),
+                          ([[1, 0, 1], [0, 0, 1]],
+                           [[1, 0, 1], [0, 0, 1], [0, 0, 1], [0, 0, 1]],
+                           np.array([1, 3]))
+                          ])
+def test_standard_scaler_sparse_sample_weights(Xw, X, sample_weight):
+    # weighted StandardScaler throughs notImplementedError when run with
+    # sample_weight
+    Xw_sparse = sparse.csr_matrix(Xw)
+    yw = np.ones(len(Xw))
+
+    scaler_w = StandardScaler(with_mean=False)
+    with pytest.raises(NotImplementedError):
+        scaler_w.fit(Xw_sparse, yw, sample_weight=sample_weight)
+
+    # but passes through when run without sample_weight
+    X_sparse = sparse.csr_matrix(X)
+    y = np.ones(len(X))
+    scaler = StandardScaler(with_mean=False)
+    scaler.fit(X_sparse, y)
+
+
 def test_standard_scaler_1d():
     # Test scaling of dataset along single axis
     for X in [X_1row, X_1col, X_list_1row, X_list_1row]:
-
         scaler = StandardScaler()
-        X_scaled = scaler.fit(X).transform(X, copy=True)
+        X_scaled = scaler.fit(X).transform(
+            X, copy=True)
```

Comment:
Why this formatting change? I believe the original code was already PEP8 compliant, no?